src/main/java/rx/internal/operators/OnSubscribeUsing.java

```diff
@@ -56,22 +59,43 @@ public void call(final Subscriber<? super T> subscriber) {
             // dispose on unsubscription
             subscriber.add(disposeOnceOnly);
             // create the observable
-            final Observable<? extends T> source = observableFactory
-            // create the observable
-                    .call(resource);
+            final Observable<? extends T> source;
+
+            try {
+                source = observableFactory
+                // create the observable
+                        .call(resource);
+            } catch (Throwable e) {
+                Throwable disposeError = dispose(disposeOnceOnly);
+                Exceptions.throwIfFatal(e);
+                Exceptions.throwIfFatal(disposeError);
+                if (disposeError != null) {
+                    subscriber.onError(new CompositeException(Arrays.asList(e, disposeError)));
+                } else {
+                    // propagate error
+                    subscriber.onError(e);
+                }
+                return;
+            }
+
             final Observable<? extends T> observable;
             // supplement with on termination disposal if requested
-            if (disposeEagerly)
+            if (disposeEagerly) {
                 observable = source
                 // dispose on completion or error
                         .doOnTerminate(disposeOnceOnly);
-            else
-                observable = source;
+            } else {
+                // dispose after the terminal signals were sent out
+                observable = source
+                        .doAfterTerminate(disposeOnceOnly)
+                        ;
```

Comment:
Just to clarify, I'm talking about the lone semicolon on line 91 that would be a perfect fit for ending the line 90.
I don't see any other lone semicolon in that file.